org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java

```diff
@@ -268,8 +269,10 @@ public CompletionItem toCompletionItem(CompletionProposal proposal, int index) {
 		proposalProvider.updateReplacement(proposal, $, '\0');
 		// Make sure `filterText` matches `textEdit`
 		// See https://github.com/eclipse/eclipse.jdt.ls/issues/1348
-		if (proposal.getKind() == CompletionProposal.TYPE_REF && $.getTextEdit() != null && $.getTextEdit().getRange() != null && $.getTextEdit().getNewText() != null) {
-			$.setFilterText($.getTextEdit().getNewText());
+		String newText = $.getTextEdit().isLeft() ? $.getTextEdit().getLeft().getNewText() : $.getTextEdit().getRight().getNewText();
+		Range range = $.getTextEdit().isLeft() ? $.getTextEdit().getLeft().getRange() : ($.getTextEdit().getRight().getInsert() != null ? $.getTextEdit().getRight().getInsert() : $.getTextEdit().getRight().getReplace());
+		if (proposal.getKind() == CompletionProposal.TYPE_REF && $.getTextEdit() != null && range != null && newText != null) {
```

Comment:
The check `$.getTextEdit() != null` at this line has no meaning. It should move up to line 272.

Reply:
Fixed.